forte/data/data_store.py

```diff
@@ -187,6 +197,76 @@ def _new_tid(self) -> int:
         r"""This function generates a new ``tid`` for an entry."""
         return uuid.uuid4().int
 
+    def _get_type_info(self, type_name: str) -> Dict[str, int]:
+        """
+        Add a new type input_entry_class into ``_type_attributes``.
+        This method will get all attributes of a type and assign index to them.
+        Args:
+            type_name (str): A string or class type representing a class.
+        Returns:
+            attr_dict (dict): The attribute dict of the given type.
+        Raises:
+            RuntimeError: when the type is not provided by ontology file and
+            dynamic import is disabled.
+        """
+        # check if type is in dictionary
+        if type_name in self._type_attributes:
+            return self._type_attributes[type_name]
+        if not self._dynamically_add_type:
+            raise RuntimeError(
+                f"{type_name} is not an existing type in current data store."
+                f"Dynamically add type is disabled."
+            )
+        # get attribute dictionary
+        attributes = self._get_entry_attributes_by_class(type_name)
+
+        attr_dict = {}
+        attr_idx = constants.ENTRY_TYPE_INDEX + 1
+        for attr_name in attributes:
+            attr_dict[attr_name] = attr_idx
+            attr_idx += 1
+
+        self._type_attributes[type_name] = {
+            "attributes": attr_dict,
+            "parent_entry": None,
+        }
+
+        return attr_dict
+
+    def _get_type_attribute_info(self, type_name: str) -> Dict[str, Any]:
```

Comment:
what's the purpose of this function? is it a weaker version of `_get_type_info`? This function also ignores dynamic typing. should we not use this?

Reply:
There are a couple of counter-arguments:
1.  This creates multiple `get_info` api calls, making it confusing which one to call.
2. Can we really ensure the dynamic-type can be done before call thing this function?
3. Even if 2 is ensured, this function will be as fast as `_get_type_info`, so what's the benefit of having this?

Reply:
@qinzzz Indeed, it seems like the only benefit here is they could provide different error messages.

Reply:
Previously I thought set and get attribute function should only get and not add new types, so i used a new function to pre vent them from adding new types. But yes, the new API does not actually provide benefits, since adding a new type if it does not exist is still a reasonable behaviour.